src/esbmc/bmc.cpp

```diff
@@ -233,7 +233,14 @@ smt_convt::resultt bmct::run_decision_procedure(
     options.get_bool_option("smt-formula-too") ||
     options.get_bool_option("smt-formula-only"))
   {
-    smt_conv.dump_smt();
+    std::string smt_formula = smt_conv.dump_smt();
+
+    // Print the SMT formula to stdout
+    if (!smt_formula.empty())
+    {
+      fprintf(stdout, "%s\n", smt_formula.c_str());
```

Comment:
You should check for the `--output` option for the destination: https://github.com/esbmc/esbmc/blob/master/src/esbmc/options.cpp#L320